tools/SetupFlow/DevHome.SetupFlow.Common/Configuration/ConfigurationFileHelper.cs

```diff
@@ -41,19 +42,14 @@ public ApplicationResult(ApplyConfigurationSetResult result)
         }
     }
 
-    private readonly StorageFile _file;
     private ConfigurationProcessor _processor;
     private ConfigurationSet _configSet;
 
-    public ConfigurationFileHelper(StorageFile file)
-    {
-        _file = file;
-    }
-
-    public async Task OpenConfigurationSetAsync()
+    public async Task OpenConfigurationSetAsync(string filePath, string content)
     {
         try
         {
+            var file = await StorageFile.GetFileFromPathAsync(filePath);

```

Comment:
What is the `filePath` used for? Would it make sense to read the file again to compare against the content passed as an argument?

Reply:
The file path is used to set the relative path used for applying the DSC configuration (if the `dsc.yaml` is using `WinGetConfigRoot` keyword), but the DSC configuration content is read from the input stream, which is passed separately and displayed on the UAC. No need to re-read from the elevated process as we are applying the snapshot that was sent from dev home process.

Reply:
Hmmmm. I guess there's really no need to read it again if we already know the contents it had. It feels weird that we could continue the process if the contents of the file changed, but it would also feel weird to stop.

Only benefit I can think of reading the file again to compare with the stream we got, is that the configuration file may have stricter permissions than the non elevated process (e.g., anyone can read, only admin can write), and comparing it could add a small additional layer of verification. Not sure if it would be worth it, though.